bundles/org.eclipse.passage.loc.report.ui/src/org/eclipse/passage/loc/report/internal/ui/jface/ExportCustomersWizard.java

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+ * Copyright (c) 2020 ArSysOp
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *      ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.loc.report.internal.ui.jface;
+
+import java.nio.file.Path;
+import java.util.Set;
+
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.jface.wizard.Wizard;
+import org.eclipse.passage.lic.products.registry.ProductRegistry;
+import org.eclipse.passage.loc.report.internal.core.CustomerStorage;
+import org.eclipse.passage.loc.report.internal.core.ExportService;
+import org.eclipse.passage.loc.report.internal.ui.i18n.ExportCustomersWizardMessages;
+import org.eclipse.passage.loc.yars.internal.api.ReportException;
+import org.eclipse.swt.program.Program;
+import org.eclipse.swt.widgets.Composite;
+
+@SuppressWarnings("restriction")
+final class ExportCustomersWizard extends Wizard {
+
+	private final ExportService export;
+	private final DataForExport data;
+	private final PreviewPage preview;
+	private final ScopePage scope;
+	private final TargetPage target;
+
+	// private final String scope = "Scope"; //$NON-NLS-1$
+
+	public ExportCustomersWizard(ProductRegistry products, CustomerStorage customers, ExportService export) {
+		this.export = export;
+		this.data = new DataForExport(this::identifiers, this::path, this::open);
+		this.preview = new PreviewPage(customers, data);
+		this.scope = new ScopePage(new Products(products, customers), preview);
+		this.target = new TargetPage(preview);
+	}
+
+	@Override
+	public void addPages() {
+		addPage(scope);
+		addPage(target);
+		addPage(preview);
+	}
+
+	@Override
+	public void createPageControls(Composite pageContainer) {
+		super.createPageControls(pageContainer);
+		scope.installInitial();
+		target.installInitial();
+	}
+
+	private Set<String> identifiers() {
+		return scope.identifiers();
+	}
+
+	private Path path() {
+		return target.path();
+	}
+
+	private boolean open() {
+		return target.open();
+	}
+
+	@Override
+	public boolean performFinish() {
+		Path file = new FileForExport(data.target()).get();
+		try {
+			export.exportCustomersForProducts(data.products(), file);
```

Comment:
I would expect `IProgressMonitor` callback and running with `IRunnableContext` from `IWizardContainer` . However, for today it may be omitted as our data is relatively small.

Reply:
Such an ability is not provided by the YARS API first, then covered by `loc.report` and thus made accessible here. [560445](https://bugs.eclipse.org/bugs/show_bug.cgi?id=560445) ticket is created to guide this work.